### ISSUE TITLE
concatenation of homogeneous matrices of nonzero degree does not preserve degree

### ISSUE THREAD

**Issue by mikestillman** [Macaulay2/M2#607]
Concatenation is not respecting degrees of matrices.
```
  S = ZZ/101[a..d]
  m = map(S^1, S^1, {{a^2}}, Degree=>2)
  isHomogeneous m
  isHomogeneous(m | m)
  isHomogeneous(m || m)
  degree m
  degree (m|m) -- should be {2}
  assert(degree m == degree (m|m))
  assert(degree m == degree(m||m))
```

**Comment by DanGrayson** [comment]
This is certainly a top level issue.  We should also make
```
degree matrix {{m,m}}
degree matrix {{m},{m}}
```
correct.

**Comment by d-torrance** [comment]
Reopening: this still reproduces on 1.26.06 — `degree(m|m)` is `{0}`, not `{2}`. I think it was closed as a duplicate of #736, but the two are distinct:  here the summand degrees agree so `ggConcatCols` does compute the right degree, which is then discarded because `map(Module,Module,Nothing,RawMatrix)` in `ringmap.m2` drops its options — whereas in #736 the degrees differ, so no degree is computed in the first place.

*AI Disclosure: I had Claude take a look at #4487, it found this issue, and then it suggested the above message.*

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Three open issues report this same defect, one per concatenation operator, and
as far as I can tell they have never been cross-referenced:

- #607, for `|` and `||`
- #736, for `|` again
- #1060, for `++`

All three still reproduce on 1.26.06-8-g34d5846039, and one contrast localizes
the fault. Multiplication *does* carry the degree; only the concatenations drop
it:

```m2
i1 : Q = QQ[t_1,t_2];

i2 : f = map(Q^1, Q^1, {{t_1}}, Degree => {1});

i3 : degree f, isHomogeneous f

o3 = ({1}, true)

i4 : degree (f*f)

o4 = {2}

i5 : degree (f||f), degree (f|f), degree (f++f)

o5 = ({0}, {0}, {0})

i6 : isHomogeneous (f||f), isHomogeneous (f|f), isHomogeneous (f++f)

o6 = (false, false, false)
```

This was also reported in the pre-GitHub `bugs/` tree, as
`bugs/dan/0-degrees-of-maps`, where Dan Grayson put it the same way:

> For other operations, such as matrix multiplication, if the degrees all differ
> by the same number (or vector), then it will be incorporated as the degree of
> the result. It is a bug that the operation `||` doesn't incorporate the degree
> into the result if both operands have the same degree [...] same for `|` and
> `++`.

Worth noting that the transcript in that file no longer demonstrates anything —
`degree (vars Q * id_(Q^2))` is `{0}` today where it was `{1}` — so the four
lines above are a fresh reproducer rather than his.

Recording it here rather than opening a fourth issue.

<sub>From [`bugs/dan/0-degrees-of-maps`](https://github.com/Macaulay2/M2/blob/388c1ff0ce30d83751dea7bc7eac77fdc1305dd7/bugs/dan/0-degrees-of-maps) in the pre-GitHub `bugs/` tree, being triaged in [#36](https://github.com/Macaulay2/M2/issues/36).  Source of truth: [`bug-triage/catalog.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/catalog.tsv).</sub>
